src/python/pants/backend/shell/target_types.py

```diff
@@ -268,6 +268,37 @@ class ShellCommandOutputsField(StringSequenceField):
         Use a trailing slash on directory names, i.e. `my_dir/`.
         """
     )
+    removal_hint = "To fix, use `output_files` and `output_directories` instead."
+    removal_version = "2.17.0.dev0"
+
+
+class ShellCommandOutputFilesField(StringSequenceField):
+    alias = "output_files"
+    required = False
+    default = ()
+    help = softwrap(
+        """
+        Specify the shell command's output files to capture.
+
+        For directories, use `output_directories`. At least one of `output_files` and
+        `output_directories` must be specified.
+        """
+    )
+
+
+class ShellCommandOutputDirectoriesField(StringSequenceField):
+    alias = "output_directories"
+    required = False
+    default = ()
+    help = softwrap(
+        """
+        Specify full directories (including recursive descendants) of output to capture from the
```

Comment:
> (including recursive descendants)

What does this mean? Will the directory include all its subdirectories? More clear wording could be:

> Specify directories of output to capture from the shell command. Subdirectories will be included recursively.